aip/general/0162.md

```diff
@@ -107,7 +116,16 @@ message CommitBookRequest {
   eight characters of a [UUID version 4 (random)](UUID4).
 - `tag` field is optional. If provided, revision created will have the tag set.
 
-### Tagging revisions
```

Comment:
Can we have a clearer way to distinguish between tags vs revisions? 
Otherwise, when the user calls 
`GetBookRevision publishers/{publisher}/books/{book}/revisions/{user_defined_tag}`, the server needs to have some heuristic to figure out whether to treat the revision as a tag or a revision_id.

Also, there can be collisions. For example, if the server assigns a revision_id "cafe" to a revision. Then the user tries to assign the "cafe" tag.

Some options are - 
1. GetBookRevision only works with revision_ids. We have a separate GetBookRevisionByTag which only works with tag names. 
2. We don't allow users to create tags that could be valid revision_ids.